src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -355,10 +359,12 @@ private boolean isSafeToDeserialize(Throwable inException) {
                 return false;
             }
 
+            // these are exceptions that JACKSON fails to deserialize
             if (inException instanceof StorageException ||
                 inException instanceof BigQueryException ||
                 inException instanceof PSQLException ||
-                inException instanceof URISyntaxException) {
+                inException instanceof URISyntaxException ||
+                inException instanceof GoogleJsonResponseException) {
```

Comment:
At this point, the only alternative we have come up with is to serialize and deserialize and catch an error before we write the state to the database. It would work, but is not exactly graceful. It also extends the window between when we catch the error and when we record the error.

So, yeah, when we come up with a more graceful method, let's use it...

Reply:
I agree it's a hassle having to update it each time we find a new exception, but to me the main problem is that if we hit a new exception in production, the actual exception is masked by the JACKSON failure to deserialize the JSON. I discussed it with Dan, who suggested logging the problematic exception JSON, so that we can at least see what happened.

I'm not sure if there's a great alternative here. We want the condition to say "if JACKSON will have a problem with this" then rewrite() it. I think the only way to do that reliably is to serialize and then immediately try to deserialize it before saving it in the stairway db. But that seems like a lot of work for every exception.

Reply:
I think it could be worth the overhead of trying to deserialize each time if it can help us to not fail on "unknown" exceptions. I created https://broadworkbench.atlassian.net/browse/DR-641 to track this.